### PR TITLE
chore(storage-browser): use Message in CreateFolder to show feedback

### PR DESCRIPTION
**Pull request by hbuchel**
#### Description of changes

- Shows a success/error Message in the CreateFolder UI.
- Makes Message dismissable

Screenshot 2024-08-16 at 12 13 46 PM

(set offline to trigger error message)
Screenshot 2024-08-16 at 12 13 07 PM

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageBrowser/Views/Controls/Message.tsx [hunk 1]
```diff
@@ -88,20 +94,22 @@ interface MessageControlProps {
 interface _MessageControl<
   T extends StorageBrowserElements = StorageBrowserElements,
 > extends Pick<T, 'Icon' | 'View'> {
-  ({ variant }: MessageControlProps): React.JSX.Element;
+  ({ variant }: MessageControlProps): React.JSX.Element | null;
```
Comment (hbuchel) [inline packages/react-storage/src/components/StorageBrowser/Views/Controls/Message.tsx:97]: Is this weird? If you dismiss the message it should be removed from the DOM

File: packages/react-storage/src/components/StorageBrowser/Views/Controls/Message.tsx [hunk 2]
```diff
@@ -88,20 +94,22 @@ interface MessageControlProps {
...
 export interface MessageControl<
   T extends StorageBrowserElements = StorageBrowserElements,
 > extends OmitElements<_MessageControl<T>, 'Container'> {
-  ({ variant }: MessageControlProps): React.JSX.Element;
+  ({ variant }: MessageControlProps): React.JSX.Element | null;
 }
+
 export const MessageControl: MessageControl = ({ variant, children }) => {
-  return (
+  const [dismissed, setDismissed] = React.useState<boolean>(false);
```
Comment (hbuchel) [inline packages/react-storage/src/components/StorageBrowser/Views/Controls/Message.tsx:107]: This is one of those local state things that I'm not sure is correct given how we want these to eventually be composable so open to any feedback.

File: packages/react-storage/src/components/StorageBrowser/Views/Controls/Message.tsx (hunk 1 above)
Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/Views/Controls/Message.tsx:97]: Think it makes sense for at least the immediate term

File: packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/CreateFolderControls.tsx [hunk 3]
```diff
@@ -15,6 +14,29 @@ export const isValidFolderName = (name: string | undefined): boolean => {
...
+    type: 'CREATE_FOLDER',
+  });
+
+  switch (result?.status) {
+    case 'SUCCESS':
+      return <Message variant="success">Folder created.</Message>;
+    case 'ERROR':
+      return (
+        <Message variant="error">
+          There was an issue creating the folder.
```
[the remark this answers is not in the document]
Comment (hbuchel) [reply packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/CreateFolderControls.tsx]: Updated in the latest to show the returned error message, or if no message is returned with ERROR result, show a default error message.

**Review by timngyn — DISMISSED** [review]
LGTM. I think the way you're using context to get the `result` inside `CreateFolderControls.tsx` makes sense

**Review by timngyn — DISMISSED** [review]

**Review by timngyn — DISMISSED** [review]

**Review by timngyn — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**